numpy/array_api/_creation_functions.py

```diff
@@ -0,0 +1,316 @@
+from __future__ import annotations
+
+
+from typing import TYPE_CHECKING, List, Optional, Tuple, Union
+
+if TYPE_CHECKING:
+    from ._typing import (
+        Array,
```

Comment:
Hi @asmeurer, I am debugging a Sphinx issue and you might know better in the typing system: I thought in normal Python interpreter `TYPE_CHECKING` is `false`, meaning `Array` et al are not imported. But throughout this module `Array` is used (ex: `Array._new()`) -- how is it possible?

Elsewhere, `Array` is imported from `_array_object` directly, instead of from `_typing`. I suspect this leads to early expansion of the `Array` type (compare [this function](https://cupy--5698.org.readthedocs.build/en/5698/reference/array_api.html#cupyx.array_api.where) and [this function](https://cupy--5698.org.readthedocs.build/en/5698/reference/array_api.html#cupyx.array_api.zeros): the former leaks the implementation detail), and I believe applying the treatment in this `_creation_functions` module would fix the problem. I just don't understand why we conditionally import here but things still work.

Reply:
Sorry nvm, I figured out my silliness: Array is imported in every function 😅